wildwood/forest.py

```diff
@@ -249,7 +250,9 @@ def fit(self, X, y, sample_weight=None):
 
         # TODO: Why only float64 ? What if the data is already binned ?
         X, y = self._validate_data(
-            X, y, dtype=[np.float32], force_all_finite=False, order="F"
+            X, y, dtype=[np.float64], force_all_finite=False, order="F"
```

Comment:
@stephanegaiffas any idea on why here we needed to change to `np.float32`, and why this would raise `ValueError: Buffer dtype mismatch, expected 'const X_DTYPE_C' but got 'float'` with `self._bin_mapper.fit_transform(X)`?

Reply:
I don't know... this is weird. We need to check the code of the Binner just to be sure that everything is okay.
But, `y` must be `float32` I think, while `X` we don't really care since it's binned and replaced by `uint8` after

Reply:
oui je suis d'accord.. je laisse `np.float64` ici pour l'instant, je ne comprends toujours pas pourquoi il y a cette erreur..

Reply:
oui je suis d'accord.. je laisse np.float64 ici pour l'instant, je ne comprends toujours pas pourquoi il y a cette erreur..